packages/admin/cms-admin/src/userPermissions/hooks/currentUser.tsx

```diff
@@ -34,7 +35,25 @@ export const CurrentUserProvider: React.FC<{
         }
     `);
 
-    if (error) throw error.message;
+    // As this Provider is very high up in the tree, don't rely on ErrorBoundary or a configured intl-Provider here
+    if (error) {
+        const isUnauthenticated = error.graphQLErrors.some(
+            (e) => e.extensions?.exception?.status === 401 || e.extensions?.code === "UNAUTHENTICATED",
+        );
+        return (
```

Comment:
With that you unmount all previously existing components and lose their local state that might include unsaved changes.

Throwing in useCurrentUser (that currently is done?) also unmounts all in the ErrorBoundary though, where all local state is lost.

Reply:
The error is thrown here in the provider, which would also unmount all children, so shouldn't change anything to the current behavior I suppose.

Reply:
Ok, probably needs improvement, but that is out of scope for this PR